primitives/src/circuit/merkle_tree/rescue_merkle_tree.rs

```diff
@@ -80,66 +86,40 @@ where
         merkle_proof: Rescue3AryMerklePathVar,
         merkle_root: Variable,
     ) -> Result<BoolVar, CircuitError> {
-        let root_var = self.compute_merkle_root(elem, &merkle_proof)?;
+        let root_var =
+            <PlonkCircuit<F> as MerkleTreeHelperGadget<RescueMerkleTree<F>>>::compute_merkle_root(
+                self,
+                elem,
+                &merkle_proof,
+            )?;
         self.is_equal(root_var, merkle_root)
     }
 
-    fn enforce_merkle_proof(
+    fn enforce_membership_proof(
         &mut self,
         elem: StandardLeafVar,
         merkle_proof: Rescue3AryMerklePathVar,
         expected_merkle_root: Variable,
     ) -> Result<(), CircuitError> {
-        let bool_val = self
-            .is_member(elem, merkle_proof, expected_merkle_root)
-            .unwrap();
+        let bool_val = <PlonkCircuit<F> as MerkleTreeGadget<RescueMerkleTree<F>>>::is_member(
+            self,
+            elem,
+            merkle_proof,
+            expected_merkle_root,
+        )
+        .unwrap();
```

Comment:
I don't see why we are using so many `.unwrap()` in non-test code. @tessico do you mind helping us remove all of them and propagate `CircuitError` properly?

Reply:
Nice catch, fixed in https://github.com/EspressoSystems/jellyfish/pull/172/commits/82ca71a9da9de693812963a3892d81613eef25d8.